lib/game.rb

```diff
@@ -1,71 +1,38 @@
 require_relative "board.rb"
-require_relative "coordinates.rb"
+require_relative "interface.rb"
+require_relative "player.rb"
 
 class Game
-  attr_reader :std_out, :std_in, :coordinates
+  attr_reader :interface, :coordinates
 
-  def initialize(std_out, std_in)
-    @std_out = std_out
-    @std_in = std_in
+  def initialize(interface:)
+    @interface = interface
   end
 
   def self.run(*args)
     Game.new(*args).run
   end
 
   def run
-    start_game
-    get_user_coordinates
-    end_game
+    start
+    play
+    finish
   end
 
   private
 
-  def start_game
-    std_out.puts game_start_message
-    set_coordinates_from_user_input
+  def start
+    interface.print_start
   end
 
-  def game_start_message
-    empty_board + "\n" + start_prompt + "\n" + instructions_prompt
+  def play
+    player = Player.new(
+      interface: interface
+    )
+    player.make_move
```

Comment:
I'm still not completely sold on this. I went back and forth around if we should set coordinates here and instantiate a player with their first move or the way I have it here.

Reply:
But I kind of like that game doesn't worry about coordinates and the player handles getting its own move.

Reply:
I also like that Game doesn't know these details. Something that might be interesting to think about is what does it mean to make a move? Right now it's to get and validate input, then... we don't do anything with it. Did we make anything?

Reply:
Following up, `make_move` feels out of place. Because it's validating the move, and getting a new move if needed, it's more of an extension of `get_move`. Could we combine those methods, or otherwise restrict the public interface to only `get_move`?

Reply:
This is a great observation! We now have one method that gets a valid move. Let me know what you think!